File: Assets/code/DFS.c

// C implementation of Depth-First Search (DFS) for a graph

#include<stdio.h>
#include<stdbool.h>

#define MAX_VERTICES 100

typedef struct {
    int edges[MAX_VERTICES][MAX_VERTICES];
    int numVertices;
    bool visited[MAX_VERTICES];
} Graph;

// Function to initialize the graph
void initializeGraph(Graph* g, int numVertices) {
    g->numVertices = numVertices;
    for (int i = 0; i < numVertices; i++) {
        for (int j = 0; j < numVertices; j++) {
            g->edges[i][j] = 0;
        }
        g->visited[i] = false;
    }
}

// Function to add an edge to the graph
void addEdge(Graph* g, int u, int v) {
    g->edges[u][v] = 1;
    g->edges[v][u] = 1; // Assuming an undirected graph
}

// Depth-First Search
void dfs(Graph* g, int vertex) {
    g->visited[vertex] = true;
    printf("%d ", vertex);

    for (int i = 0; i < g->numVertices; i++) {
        if (g->edges[vertex][i] == 1 && !g->visited[i]) {
            dfs(g, i);
        }
    }
}

int main() {
    Graph g;
    int numVertices = 7; // Change this according to the graph size
    initializeGraph(&g, numVertices);

    // Add edges to the graph (example)
    addEdge(&g, 0, 1);
    addEdge(&g, 0, 2);
    addEdge(&g, 1, 3);
    addEdge(&g, 1, 4);
    addEdge(&g, 2, 5);
    addEdge(&g, 2, 6);

    printf("Depth-First Traversal starting from vertex 0: ");
    dfs(&g, 0);

    return 0;
}
              
            

File: Assets/code/DFS.java
import java.util.*;

public class DFS {
    private int numVertices;
    private boolean[][] edges;
    private boolean[] visited;

    public DFS(int numVertices) {
        this.numVertices = numVertices;
        edges = new boolean[numVertices][numVertices];
        visited = new boolean[numVertices];
    }

    public void addEdge(int u, int v) {
        edges[u][v] = true;
        edges[v][u] = true; // Assuming an undirected graph
    }

    public void dfs(int vertex) {
        visited[vertex] = true;
        System.out.print(vertex + " ");

        for (int i = 0; i < numVertices; i++) {
            if (edges[vertex][i] && !visited[i]) {
                dfs(i);
            }
        }
    }

    public static void main(String[] args) {
        DFS g = new DFS(7); // Change this according to the graph size

        // Add edges to the graph (example)
        g.addEdge(0, 1);
        g.addEdge(0, 2);
        g.addEdge(1, 3);
        g.addEdge(1, 4);
        g.addEdge(2, 5);
        g.addEdge(2, 6);

        System.out.print("Depth-First Traversal starting from vertex 0: ");
        g.dfs(0);
    }
}
